mirgecom/diffusion.py

```diff
@@ -49,22 +48,26 @@ def _u_flux(discr, alpha, q_tpair):
     return discr.project(q_tpair.dd, "all_faces", flux_weak)
 
 
-def diffusion_operator(discr, alpha, u):
+def diffusion_operator(discr, alpha, u_boundaries, q_boundaries, u):
     r"""
     Compute the diffusion operator.
 
     The diffusion operator is defined as
     $\nabla\cdot(\alpha\nabla u)$, where $\alpha$ is the diffusivity and
     $u$ is a scalar field.
 
-    Currently sets $u = 0$ on the boundaries.
-
     Parameters
     ----------
     discr: grudge.eager.EagerDGDiscretization
         the discretization to use
     alpha: float
         the (constant) diffusivity
+    u_boundaries:
+        dictionary (or object array of dictionaries) of boundary functions for *u*,
+        one for each valid btag
+    q_boundaries:
+        dictionary (or object array of dictionaries) of boundary functions for
+        :math:`q = \sqrt{\alpha}\nabla u`, one for each valid btag
```

Comment:
Would it make sense to expose Dirichlet and Neumann (and perhaps Robin) as the classical second-order BCs, rather than expecting users to piece together their own out of boundary states for `u` and `q`? (Honest question: I'm not presuming the answer to this is yes.)

Reply:
I made an attempt... is this kind of what you had in mind API-wise?